andela/firestar-backend
# #167730746 Profile Settings

**What does this PR do?**
Once registration is complete, the user profile is created and User is allowed to edit profile

**Description of Task to be completed?**
- N/A

**How should this be manually tested?**
- git clone this repo https://github.com/andela/firestar-backend.git
- cd to the project
- Checkout to the branch `ft-profile-setting-167730746`
- Create two databases name in your postgres database. e.g. (`barefoot and barefoot_test`)
- Create a .env file and set, `JWT_SECRET` and DB variables as specified in the .env.example file
- run npm install
- run `npm run migrate:dev` to generate Users table then
Then
-  Now use postman and go to `localhost:PORT/api/v1/users/auth/register` using a POST request with a body of
```
{
 "email": "exadmple@gmail.com",
 "password": "firestar2019@K",
 "firstName": "adewale",
 "lastName": "olaoye"
}
```

- If successful you would see a response containing the token, *copy* the token which will be used on `GET` and `PATCH` route

```
To GET user details
make a `GET` request to `localhost:PORT/api/v1/users/1/profile`
If successful, you would see a response containing the details of the user.
```
```
To UPDATE User details
copy the token generated by the `GET` request 
set the token in the header as Authorization with the value `Bearer tokenFromTheResponseBody`
make a `PATCH` request to `localhost:PORT/api/v1/users/1/profile`
{
     "firstName": "Rasheed",
     "lastName: "Alimi",
     "company": "Firestar"
}
```
**Any background context you want to provide?**
N/A

**What are the relevant pivotal tracker stories?**
[#167730746](https://www.pivotaltracker.com/story/show/167730746)

**Screenshots or gifs (if appropriate)**
![Screenshot (21)](https://user-images.githubusercontent.com/44645460/64913580-7d31c700-d73a-11e9-8cb4-306b6d87f43b.png)

**Questions:**
- N/A
